torch/distributed/distributed_c10d.py

```diff
@@ -1307,6 +1309,27 @@ def batch_isend_irecv(p2p_op_list):
     return reqs
 
 
+def exception_handler(func):
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
+        try:
+            return func(*args, **kwargs)
+        except Exception as error:
+            error_logger = _get_or_create_logger()
+            error_msg_dict = {
+                "func_name": f"{func.__name__}",
+                "args": f"{args}, {kwargs}",
+                "world_size": f"{get_world_size()}",
```

Comment:
It looks like some tests are using the direct initialization python initialization of ProcessGroupGloo / ProcessGroupNCCL (https://github.com/pytorch/pytorch/blob/master/test/distributed/test_c10d_gloo.py#L208). This is technically unsupported as it skips the state dictionaries that we keep to track `default group`, `backend`, etc. which is set during `init_process_group`. However, since we have these tests in CI, we see failures like "Default process group has not been initialized, please make sure to call init_process_group." (e.g. https://github.com/pytorch/pytorch/actions/runs/3318590784/jobs/5482967896)

Rather than updating all the tests to use `init_process_group` in this PR. We should guard `get_world_size()`, `get_rank()`, and `get_backend()` with `is_initialized()`.